arbos/block_processor.go

```diff
@@ -193,7 +193,8 @@ func ProduceBlockAdvanced(
 	chainConfig := chainContext.Config()
 
 	header := createNewHeader(lastBlockHeader, l1Info, arbState, chainConfig)
-	signer := types.MakeSigner(chainConfig, header.Number, header.Time)
+
+	signer := types.VersionedArbitrumSigner(chainConfig, header.Number, header.Time, arbState.ArbOSVersion())
```

Comment:
move that inside the tx processing loop, in case the first transaction upgrades arbos version
(BTW - here you should indeed look at arboState and not the header, just as you already do, for that exact reason)

Reply:
Done.